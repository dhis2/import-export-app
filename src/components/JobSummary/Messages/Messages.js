import i18n from '@dhis2/d2-i18n'
import {
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
    TableRowHead,
    TableCellHead,
} from '@dhis2/ui'
import React from 'react'
import { FormField } from '../../index'
import { messagesPropType } from '../helper'

const Messages = ({ messages }) => {
    if (messages.length == 0) {
        return null
    }
    return (
        <FormField
            label={`${i18n.t('Messages')}`}
            dataTest="job-summary-messages"
            name="messages"
        >
            <Table>
                <TableHead>
                    <TableRowHead>
                        <TableCellHead>{i18n.t('UID')}</TableCellHead>
                        <TableCellHead>{i18n.t('Type')}</TableCellHead>
                        <TableCellHead>{i18n.t('Property')}</TableCellHead>
                        <TableCellHead>{i18n.t('Message')}</TableCellHead>
                    </TableRowHead>
                </TableHead>
                <TableBody>
                    {messages.map((m, i) => (
                        <TableRow key={`job-summary-messages-${i}`}>
                            <TableCell>{m.uid}</TableCell>
                            <TableCell>{m.type}</TableCell>
                            <TableCell>{m.property}</TableCell>
                            <TableCell>{m.message}</TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </FormField>
    )
}

Messages.propTypes = {
    messages: messagesPropType.isRequired,
}

export { Messages, messagesPropType }
