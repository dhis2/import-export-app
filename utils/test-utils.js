import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
