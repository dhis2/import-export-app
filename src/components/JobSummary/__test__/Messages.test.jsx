import React from 'react'
import { render } from 'test-utils'

import { Messages } from '../Messages/Messages.jsx'

const props = {
    messages: [
        {
            uid: 'abc',
            type: 'DefaultGmlImportService',
            property: 'def',
            message: 'message one',
        },
        {
            uid: 'xyz',
            type: 'SndGmlImportService',
            property: 'def',
            message: 'message one',
        },
    ],
}

it(`matches snapshot`, () => {
    const { asFragment } = render(<Messages {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
