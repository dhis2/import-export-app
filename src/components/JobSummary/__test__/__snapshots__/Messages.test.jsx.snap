// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-summary-messages"
  >
    <span
      class="label"
    >
      Messages
    </span>
    <table
      class="jsx-2430604489 "
      data-test="dhis2-uicore-table"
    >
      <thead
        data-test="dhis2-uicore-tablehead"
      >
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerowhead"
        >
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            UID
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Type
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Property
          </th>
          <th
            class="jsx-1796049005 "
            data-test="dhis2-uicore-tablecellhead"
          >
            Message
          </th>
        </tr>
      </thead>
      <tbody
        data-test="dhis2-uicore-tablebody"
      >
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            abc
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            DefaultGmlImportService
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            def
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            message one
          </td>
        </tr>
        <tr
          class="jsx-3124144591 zebraStriping"
          data-test="dhis2-uicore-tablerow"
        >
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            xyz
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            SndGmlImportService
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            def
          </td>
          <td
            class="jsx-996330601 "
            data-test="dhis2-uicore-tablecell"
          >
            message one
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
